lib/presenter.rb

```diff
@@ -11,6 +11,22 @@ def display_board(board)
     BOARD
   end
 
+  def draw_outcome
+    <<~DRAW
+
+      ---------------Draw!---------------
+
+    DRAW
+  end
+
+  def win_outcome(token)
+    <<~WIN
+
+      ---------------#{token} wins!---------------
+
+    WIN
+  end
```

Comment:
The public interface of the presenter has widened now, and each method expects a different argument (or no argument in the case of `draw_outcome`).

The implication of this is that we are expecting the client of the presenter to know how to use it correctly. Can you think of another design alternative that would simplify this and keep the logic contained within the presenter? 🤔

Reply:
dependency injection?